docs/manual/src/udl/errors.md

```diff
@@ -42,3 +42,85 @@ interface ArithmeticError {
   IntegerOverflow(u64 a, u64 b);
 };
 ```
+
+## Interfaces as errors
+
+** This is not avaiable in all bindings **
+
+It's possible to use an `interface` (ie, a rust struct impl) as an error;
+the thrown object will have methods instead of fields.
+This can be particularly useful when working with `anyhow` style errors, where
+an enumeration can't represent the various error states.
+
+In your UDL:
+```
+namespace error {
+  [Throws=MyError]
+  void bail(string message);
+}
+
+[Traits=(Debug)]
+interface MyError {
+  string message();
+};
+```
+and Rust:
+```rs
+#[derive(Debug)]
+pub struct MyError {
+    e: anyhow::Error,
+}
```

Comment:
Would it be possible to return `anyhow::Error` directly, maybe together with a crate-local trait that supplies the `message()` method?